src/source/src/parser/mod.rs

```diff
@@ -189,31 +190,31 @@ impl SourceStreamChunkRowWriter<'_> {
         &mut self,
         mut f: impl FnMut(&SourceColumnDesc) -> Result<A::Output>,
     ) -> Result<WriteGuard> {
-        // The closure `f` may fail so that a part of builders were appended incompletely.
-        // Loop invariant: `builders[0..appended_idx)` has been appended on every iter ended or loop
-        // exited.
-        let mut appended_idx = 0;
+        let mut modify_col = vec![];
 
         self.descs
             .iter()
             .zip_eq(self.builders.iter_mut())
             .enumerate()
             .try_for_each(|(idx, (desc, builder))| -> Result<()> {
+                if desc.is_timestamp {
```

Comment:
I mean we can check if the column name starts with `_rw` and fill the timestamp here.